### PR TITLE
TC1 Dynamic Weight Integration

### PR DESCRIPTION
**Pull request by James-Featherston**
**Description**
This PR incorporates the dynamic weights from the last PR into the main recommendation algorithm. It also promotes fairness by changing the way the sport preference is calculated. It divides the sport count clicked by the max sport clicks in the sportBuckets instead of by the total clicks for all sports so that a click on a sport can grow as fast as a click on an event in the near future. It also sets a max count that an event can have so that as the user clicks on more and more events, the weights will stay volatile.

**Milestones**
This works towards TC1.

**Resources**
N/A

**Test Plan**
To test, I printed out the weights in the terminal to make sure that everything was changing as expected. I also started fresh with a new user to make sure that there was not initialization edge cases I missed. The image shows the weights changing on different clicks.
image

### CODE REVIEW

File: server/config.js [hunk 1]
```diff
@@ -49,17 +49,20 @@ calculation in the event recommendation algorithm
 // Used to calculate the likely hood an event was cliked on
```
Comment (yanhao-li) [inline server/config.js:49]:
```suggestion
// Used to calculate the likely hood an event was clicked on
```

File: server/recommendations/eventWeights.js [hunk 2]
```diff
@@ -157,4 +163,11 @@ const updateRecommendationData = async (userId, newData) => {
   });
   return result;
 };
+
+const refreshRecommendationData = async (userId) => {
```
Comment (yanhao-li) [inline server/recommendations/eventWeights.js:167]: dead code? where is it used?

**Review by yanhao-li — APPROVED** [review]
